lib/tran_sock.c

```diff
@@ -762,26 +787,35 @@ tran_sock_send_msg(vfu_ctx_t *vfu_ctx, uint16_t msg_id,
               struct vfio_user_header *hdr,
               void *recv_data, size_t recv_len)
 {
-    return tran_sock_msg(vfu_ctx->conn_fd, msg_id, cmd, send_data, send_len,
+    tran_sock_t *ts = vfu_ctx->tran_data;
+
+    return tran_sock_msg(ts->conn_fd, msg_id, cmd, send_data, send_len,
                          hdr, recv_data, recv_len);
 }
 
 static void
 tran_sock_detach(vfu_ctx_t *vfu_ctx)
 {
-    if (vfu_ctx->conn_fd != -1) {
-        (void) close(vfu_ctx->conn_fd);
-        vfu_ctx->conn_fd = -1;
+    tran_sock_t *ts = vfu_ctx->tran_data;
+
+    if (ts->conn_fd != -1) {
+        (void) close(ts->conn_fd);
```

Comment:
AFAIK we should be handling `EINTR` here, for now let's at least log a warning?

Reply:
added FIXMEs for now